test_scripts/API/VehicleData/common.lua

```diff
@@ -636,4 +502,672 @@ function m.setAppVersion(pParamVersion, pOperator)
   actions.app.getParams().syncMsgVersion.patchVersion = patch
 end
 
+--[[ @getKeyByValue: Get key from table by defined value
+--! @parameters:
+--! pTbl: table for lookup
+--! pValue: value for lookup
+--! @return: key
+--]]
+function m.getKeyByValue(pTbl, pValue)
+  for k, v in pairs(pTbl) do
+    if v == pValue then return k end
+  end
+  return nil
+end
+
+--[[ Params Generator Functions ]]------------------------------------------------------------------
+
+--[[ @getParamsValidDataTestForRequest: Provide parameters for processing valid sequence for 'GetVehicleData' request
+--! @parameters:
+--! pGraph: graph with structure of parameters
+--! @return: table with parameters
+--]]
+local function getParamsValidDataTestForRequest(pGraph)
+  local request = { [paramName] = true }
+  local hmiResponse = tdg.getParamValues(pGraph)
+  local mobileResponse = utils.cloneTable(hmiResponse)
+  mobileResponse.success = true
+  mobileResponse.resultCode = "SUCCESS"
+  local params = {
+    mobile = {
+      name = rpc,
+      request = request,
+      response = mobileResponse
+    },
+    hmi = {
+      name = m.rpcHMIMap[rpc],
+      request = request,
+      response = hmiResponse
+    }
+  }
+  return params
+end
+
+--[[ @getParamsInvalidDataTestForRequest: Provide parameters for processing invalid sequence for 'GetVehicleData' request
+--! @parameters:
+--! pGraph: graph with structure of parameters
+--! @return: table with parameters
+--]]
+local function getParamsInvalidDataTestForRequest(pGraph)
+  local request = { [paramName] = true }
+  local hmiResponse = tdg.getParamValues(pGraph)
+  local params = {
+    mobile = {
+      name = rpc,
+      request = request,
+      response = { success = false, resultCode = "GENERIC_ERROR" }
```

Comment:
Shouldn't the response here be `INVALID_DATA`

Reply:
SDL usually responds with `INVALID_DATA` in case some request from Mobile has incorrect data.
However in our case we have incorrect data sent from HMI. So SDL should reply `GENERIC_ERROR`.

Please find a requirement: 
> In case
HMI sends invalid response by any reason that SDL must transfer to mobile app
SDL must:
log the error internally
respond GENERIC_ERROR (success:false, info: "Invalid message received from vehicle") to mobile app